new-log-viewer/src/services/decoders/JsonlDecoder.ts

```diff
@@ -24,61 +29,95 @@ import LogbackFormatter from "../formatters/LogbackFormatter";
 class JsonlDecoder implements Decoder {
     static #textDecoder = new TextDecoder();
 
-    #logLevelKey: string = "level";
+    #dataArray: Nullable<Uint8Array>;
+
+    readonly #logLevelKey: string;
+
+    readonly #timestampKey: string;
+
+    #logEvents: JsonLogEvent[] = [];
 
-    #logEvents: JsonObject[] = [];
+    #isFiltered: boolean = false;
+
+    #filteredLogEventIndices: number[] = [];
 
     #invalidLogEventIdxToRawLine: Map<number, string> = new Map();
 
-    // @ts-expect-error #fomatter is set in the constructor by `setDecoderOptions()`
     #formatter: Formatter;
 
     /**
      * @param dataArray
      * @param decoderOptions
      * @throws {Error} if the initial decoder options are erroneous.
      */
-    constructor (dataArray: Uint8Array, decoderOptions: JsonlDecoderOptionsType) {
-        const isOptionSet = this.setDecoderOptions(decoderOptions);
-        if (false === isOptionSet) {
-            throw new Error(
-                `Initial decoder options are erroneous: ${JSON.stringify(decoderOptions)}`
-            );
-        }
-        this.#deserialize(dataArray);
+    constructor (dataArray: Uint8Array, decoderOptions: JsonlDecoderOptions) {
+        this.#dataArray = dataArray;
+        this.#logLevelKey = decoderOptions.logLevelKey;
+        this.#timestampKey = decoderOptions.timestampKey;
+        this.#formatter = new LogbackFormatter(decoderOptions);
     }
 
     getEstimatedNumEvents (): number {
         return this.#logEvents.length;
     }
 
-    buildIdx (beginIdx: number, endIdx: number): Nullable<LogEventCount> {
-        // This method is a dummy implementation since the actual deserialization is done in the
-        // constructor.
+    getFilteredLogEventIndices (): number[] {
+        return this.#filteredLogEventIndices;
+    }
 
-        if (0 > beginIdx || this.#logEvents.length < endIdx) {
-            return null;
+    setLogLevelFilter (logLevelFilter: LogLevelFilter): boolean {
+        this.#filterLogs(logLevelFilter);
+        this.#isFiltered = Boolean(logLevelFilter);
```

Comment:
Following the discussion at https://github.com/y-scope/yscope-log-viewer/pull/63/files#r1768464251 , I feel this can be more readable:
```suggestion
        this.#isFiltered = null !== logLevelFilter;
```